ChibiOS: fix I2C transactions

lucasdemarchi opened the pull request:
Since when we start calling i2cStop() after each transfer, the I2C bus was malfunctioning because we disable the device before it has time to complete the last ack (when receiving data) or the stop condition.

I caught this bug because the device I'm working with require the stop condition and doesn't work with repeated starts. I guess this is not a problem for most of other devices because they work normally with repeated starts and don't care about the NACKs  from the master controller.

Trace the bus before this change:
![tfmini-stm32-request-version](https://user-images.githubusercontent.com/31864/58314200-de1c4480-7dc3-11e9-85eb-5f08fe624bb8.png)
![tfmini-stm32-response-version](https://user-images.githubusercontent.com/31864/58314206-e1173500-7dc3-11e9-9370-98dd6ca80b8e.png)

Trace after the change:
![i2c-with-stop](https://user-images.githubusercontent.com/31864/58314226-ed9b8d80-7dc3-11e9-8f8f-559039e831e2.png)


This was tested with a CUAVv5, i.e. a stm32f7 that uses the I2Cv2 driver in ChibiOS. I also changed the I2Cv1 and I2Cv3 with similar changes to contemplate the other boards we support, but those are only compile-tested.

Thanks to Sergey who helped me debug the issue with his pixracer: https://discuss.ardupilot.org/t/pixracer-does-not-work-with-tfmini-i2c-lidar-chibios/41261/7

lucasdemarchi commented:
Confirmed to be working on pixracer: https://discuss.ardupilot.org/t/pixracer-does-not-work-with-tfmini-i2c-lidar-chibios/41261/7

@tridge could you test on pixhawk6 to cover the I2Cv3 driver?

bys1123 commented:
![image](https://user-images.githubusercontent.com/4748411/58763391-a44df980-858c-11e9-901e-7c79d52aac0b.png)
Thanks, tested on Pixhawk4, works fine!

tridge commented:
@lucasdemarchi sorry for the long delay on review. I'm happy with the approach and I've tested this on F4, F7 and H7.
I rebased on master, plus I fixed up some formatting errors for the I2Cv3 driver in ChibiOS (the file was converted to unix line endings, which would have made future rebases harder)
Thanks very much for this work!

lucasdemarchi commented:
> I rebased on master, plus I fixed up some formatting errors for the I2Cv3 driver in ChibiOS (the file was converted to unix line endings, which would have made future rebases harder)

yeah, I was using unix2dos to fixup while committing since I don't really wanted to change my editor's default. Then I fixed a mistake and forgot to call unix2dos again. Thanks.

lucasdemarchi commented:
@tridge do you think it would be acceptable to backport this to latest stable? I can work on that.

tridge commented:
yes, I think a backport would be good, thanks!
